### PR TITLE
Add documentation for system image

### PR DESCRIPTION
**Pull request by jkozdon**
I have been asked about this a few times in the past couple days, so thought it should be in the docs.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by jkozdon** [comment]
https://clima.github.io/ClimateMachine.jl/previews/PR1518/DevDocs/SystemImage/

File: docs/src/DevDocs/SystemImage.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+in the `@__DIR__` directory (i.e., the directory in which the script is located)
+
+Called with a single argument the system image will be placed in the path
+specified by the argument (relative to the callers path)
+
+Called with a specified systemimg path and `true`, the system image will
+compile the climate machine package module (useful for CI). This option should
+not be used when actually developing the climate machine package; see the
+[drawback](https://julialang.github.io/PackageCompiler.jl/dev/sysimages/#Drawbacks-to-custom-sysimages-1)
+from the `PackageCompiler` repository.
```
Comment (charleskawczynski) [inline docs/src/DevDocs/SystemImage.md:20]:
```suggestion
If called with

 - **no arguments**: will create the system image `ClimateMachine.so`
in the `@__DIR__` directory (i.e., the directory in which the script is located)

 - **a single argument**: the system image will be placed in the path
specified by the argument (relative to the callers path)

 - **a specified systemimg path and `true`**: the system image will
compile the climate machine package module (useful for CI). This option should
not be used when actually developing the climate machine package; see the
[drawback](https://julialang.github.io/PackageCompiler.jl/dev/sysimages/#Drawbacks-to-custom-sysimages-1)
from the `PackageCompiler` repository.
```

**Review by charleskawczynski — COMMENTED** [review]
Definitely nice to have this. I added a suggestion on some formatting

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**